hudi-common/src/main/java/org/apache/hudi/common/model/OverwriteWithLatestAvroPayload.java

```diff
@@ -49,7 +49,7 @@ public OverwriteWithLatestAvroPayload(Option<GenericRecord> record) {
   @Override
   public OverwriteWithLatestAvroPayload preCombine(OverwriteWithLatestAvroPayload another) {
     // pick the payload with greatest ordering value
-    if (another.orderingVal.compareTo(orderingVal) > 0) {
+    if (another.orderingVal.compareTo(orderingVal) >= 0) {
       return another;
```

Comment:
The original logic is actually right, because the passed in `another` is the old existing value.

Reply:
> I actually have a similar PR #3267 but have no time to finish it yet, you can take it.
OK, thanks I will close this PR